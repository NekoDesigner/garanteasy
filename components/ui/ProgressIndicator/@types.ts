/**
 * {{ElementComponent}} - Generated by Garanteasier CLI
 */

export interface IProgressIndicatorProps {
  style?: object;
  testID?: string;
  value: number;
}
