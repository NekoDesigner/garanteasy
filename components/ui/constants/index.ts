const SIZES = {
  icon: {
    s: 20,
    xs: 16,
  },
  padding: {
    xxs: 4,
    xs: 8,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 40,
    xxxl: 48,
  },
  font: {
    familly: {
      default: "Ubuntu" as const,
    },
    weight: {
      semiBold: "700" as const,
    },
    m: 16,
  },
};

const COLORS = {
  primary: "#1A1770",
  emerald: "#046C63",
  danger: "#C82828",
  light: "#FFFFFF",
  grey: "#EBEBEB",
  blueDarker: "#050426",
};

export { SIZES, COLORS };
