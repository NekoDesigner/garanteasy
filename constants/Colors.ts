export const COLORS = {
  primary: "#1A1770",
  emerald: "#046C63",
  danger: "#C82828",
  light: "#FFFFFF",
  grey: "#EBEBEB",
  blueDarker: "#050426",
  success: "#7ABA8E",
  warning: "#F4B286",
  dangerLight: "#F86068",
};