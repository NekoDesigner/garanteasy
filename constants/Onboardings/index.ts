export * from "./Onboardings";
export * from "./@types";
export * from "./screens";
