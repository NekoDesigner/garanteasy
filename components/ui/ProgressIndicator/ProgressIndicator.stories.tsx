/**
 * ProgressIndicator - Generated by Garanteasier CLI
 */

import React from "react";
import { View, Text } from "react-native";
import ProgressIndicator from "./index";

export default {
  title: 'ProgressIndicator',
  component: ProgressIndicator
};

function TemplateProgressIndicatorAnimationComponent() {
  const [progress, setProgress] = React.useState(1);
  React.useEffect(() => {
    const interval = setInterval(() => {
      setProgress((prev) => (prev > 0 ? prev - 0.01 : 1));
    }, 100);
    return () => clearInterval(interval);
  }, []);
  return (
    <View style={{ width: '60%', padding: 8, gap: 16 }}>
      <Text>Progress Animation {Math.floor(progress * 100)}%</Text>
      <ProgressIndicator value={progress} />
    </View>
  );
}

export const Default = () => (
  <View style={{ flex: 1, padding: 16, gap: 24, flexDirection: 'column', alignItems: 'center', justifyContent: 'center' }}>
    <TemplateProgressIndicatorAnimationComponent />
  </View>
);