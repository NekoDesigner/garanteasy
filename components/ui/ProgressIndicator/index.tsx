/**
 * ProgressIndicator - Generated by Garanteasier CLI
 */

import React from "react";
import { View } from "react-native";
import { IProgressIndicatorProps } from "./@types";
import ProgressIndicatorStyles from "./styles";

const ProgressIndicator: React.FC<IProgressIndicatorProps> = ({ style, testID = 'progressindicator', value, ...props }) => {
  const progressIndicatorStyles = ProgressIndicatorStyles({ percent: value });
  return (
    <View style={[progressIndicatorStyles.container, style]} testID={testID} {...props} />
  );
};

export default ProgressIndicator;
