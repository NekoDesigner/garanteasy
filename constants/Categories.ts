import { IChipsProps } from "../components/ui/Chips/@types";

export const CATEGORIES_BASE: IChipsProps[] = [
  { label: 'Électroménagé', category: 'hoursehold-electricals', showIcon: true, onPress: () => {} },
  { label: 'Petit éléctroménagé', category: 'small-electricals', showIcon: true, onPress: () => {} },
  { label: 'Bricolage', category: 'diy', showIcon: true, onPress: () => {} },
  { label: 'Jardin', category: 'garden', showIcon: true,  onPress: () => {} },
  { label: 'Mode', category: 'fashion', showIcon: true, onPress: () => {} },
  { label: 'Multimédia', category: 'multimedia', showIcon: true, onPress: () => {} },
  { label: 'Autre', category: 'other', showIcon: false, onPress: () => {} },
];