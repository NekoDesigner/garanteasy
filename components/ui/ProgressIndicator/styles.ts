/**
 * {{ElementComponent}} - Generated by Garanteasier CLI
 */

import { StyleSheet } from 'react-native';
import { COLORS, SIZES } from '../../../constants';

export interface IProgresIndicatorStyle {
  percent: number;
}

const styles = function ({ percent }: IProgresIndicatorStyle) {
  function getColorStyles(amount: number): string {
    if (percent <= 1 && percent >= 0.5) return COLORS.success;
    if (percent < 0.5 && percent >= 0.25) return COLORS.warning;
    return COLORS.dangerLight;
  }
  return StyleSheet.create({
    container: {
      borderRadius: SIZES.radius.full,
      height: 8,
      width: `${percent * 100}%`,
      backgroundColor:  getColorStyles(percent),
  }
}); };

export default styles;
