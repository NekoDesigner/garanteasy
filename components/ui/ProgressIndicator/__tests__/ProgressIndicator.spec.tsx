/**
 * ProgressIndicator - Generated by Garanteasier CLI
 */

import { render } from "@testing-library/react-native";
import React from "react";
import { COLORS } from "../../../../constants";
import ProgressIndicator from "../index";

describe("ProgressIndicator Components", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

    test('renders correctly', () => {
      const tree = render(<ProgressIndicator value={1} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

  it(`should render with ${COLORS.success} color when value is 1`, () => {
    const { getByTestId } = render(<ProgressIndicator value={1} />);
    const progressBar = getByTestId('progressindicator');
    expect(progressBar.props.style[0].backgroundColor).toBe(COLORS.success);
  });

  it(`should render with ${COLORS.warning} color when value is lower than 0.5`, () => {
    const { getByTestId } = render(<ProgressIndicator value={0.4} />);
    const progressBar = getByTestId('progressindicator');
    expect(progressBar.props.style[0].backgroundColor).toBe(COLORS.warning);
  });

  it(`should render with ${COLORS.dangerLight} color when value is lower than 0.25`, () => {
    const { getByTestId } = render(<ProgressIndicator value={0.2} />);
    const progressBar = getByTestId('progressindicator');
    expect(progressBar.props.style[0].backgroundColor).toBe(COLORS.dangerLight);
  });
});