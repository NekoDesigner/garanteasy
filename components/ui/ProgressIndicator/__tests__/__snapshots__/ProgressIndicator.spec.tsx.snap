// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator Components renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#7ABA8E",
        "borderRadius": 9999,
        "height": 8,
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="progressindicator"
/>
`;
